src/handwritten/task.py

```diff
@@ -674,21 +619,30 @@ def read(self, number_of_samples_per_channel=NUM_SAMPLES_UNSET,
                         return data[:samples_read]
             else:
                 data = numpy.zeros(array_shape, dtype=numpy.float64)
-                samples_read = _read_analog_f_64(
-                    self._handle, data, number_of_samples_per_channel, timeout)
+                _, samples_read = self._interpreter.read_analog_f64(
+                    self._handle, number_of_samples_per_channel, timeout, 
+                    FillMode.GROUP_BY_CHANNEL.value, data)
 
         # Digital Input or Digital Output
         elif (read_chan_type == ChannelType.DIGITAL_INPUT or
                 read_chan_type == ChannelType.DIGITAL_OUTPUT):
             if self.in_stream.di_num_booleans_per_chan == 1:
                 data = numpy.zeros(array_shape, dtype=bool)
-                samples_read = _read_digital_lines(
-                    self._handle, data, number_of_samples_per_channel, timeout
-                    ).samps_per_chan_read
+                _, samps_per_chan_read, num_bytes_per_samp = self._interpreter.read_digital_lines(
+                    self._handle, number_of_samples_per_channel, timeout, 
+                    FillMode.GROUP_BY_CHANNEL.value, data)
+
+                ReadDigitalLinesReturnData = (
+                        collections.namedtuple(
+                            'ReadDigitalLinesReturnData',
+                            ['samps_per_chan_read', 'num_bytes_per_samp']))
+
+                samples_read = ReadDigitalLinesReturnData(samps_per_chan_read, num_bytes_per_samp).samps_per_chan_read
```

Comment:
IMO, creating the namedtuple type in every function call is an anti-pattern. We should create namedtuple types at file scope. (Also, we should use `typing.NamedTuple`.)

In this case, there is no longer a need for a namedtuple.

```suggestion
                _, samples_read, _ = self._interpreter.read_digital_lines(
                    self._handle, number_of_samples_per_channel, timeout, 
                    FillMode.GROUP_BY_CHANNEL.value, data)
```

Reply:
Removed the named tuple and assigned `samples_read` to the return value directly.